dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java

```diff
@@ -204,4 +216,46 @@ public String scan(InputStream fileInput) throws IOException {
 
     }
 
+    public String checkFilesIntegrity() {
```

Comment:
IMO dedicated model class as result type would be much better. Not much can be done further with string.